sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -230,7 +230,7 @@ object LogicalPlanIntegrity {
       // NOTE: we still need to filter resolved expressions here because the output of
       // some resolved logical plans can have unresolved references,
       // e.g., outer references in `ExistenceJoin`.
-      p.output.filter(_.resolved).map { a => (a.exprId, a.dataType) }
+      p.output.filter(_.resolved).map { a => (a.exprId, a.dataType.asNullable) }
```

Comment:
@maropu We can eliminate cast for complex types that are compatible (only nullability is different), so the previous logic could fail valid queries.

Reply:
The view tests fail without this change. It's a test only thing that we don't need to backport, so I didn't spend time putting this into a separate PR with tests.

Reply:
got it. thanks.